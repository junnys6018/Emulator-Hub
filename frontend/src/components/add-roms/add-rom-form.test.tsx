/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import renderer from 'react-test-renderer';
import AddRomForm from './add-rom-form';
import 'jest-fetch-mock';

jest.mock('@/src/storage/game-data', () => {
    const useGameMetaData = jest.fn();
    useGameMetaData.mockReturnValue([undefined, jest.fn()]);
    return {
        __esModule: true,
        useGameMetaData,
    };
});
jest.mock('@/src/storage/storage');
jest.mock('../util/alert');
jest.mock('../util/message');

test('<AddRomForm /> snapshot', () => {
    fetchMock.mockResponseOnce('mock');
    const file = new File([], 'test');
    const tree = renderer
        .create(<AddRomForm id={0} initialName="" inititalConsole="NES" file={file} onDelete={() => {}} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
