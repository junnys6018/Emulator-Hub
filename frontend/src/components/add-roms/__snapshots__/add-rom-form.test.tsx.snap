// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddRomForm /> snapshot 1`] = `
<form
  className="flex flex-col self-center rounded-4xl w-80 bg-primary-800 p-5"
  onSubmit={[Function]}
>
  <div
    className="bg-cover bg-center bg-blend-multiply rounded-3xl bg-primary-900 h-48 mb-3 flex items-center justify-center"
  >
    <label
      className="btn-secondary h-10 px-2.5 text-primary-100"
      htmlFor="image-0"
    >
      <input
        accept="image/png, image/jpeg"
        className="hidden"
        id="image-0"
        name="image-0"
        onChange={[Function]}
        type="file"
      />
      Choose Image
    </label>
  </div>
  <div
    className="flex items-center mb-3.5"
  >
    <label
      className="font-semibold mr-auto"
      htmlFor="name-0"
    >
      Name*
    </label>
  </div>
  <input
    className="appearance-none w-full h-11 px-2 mb-5 flex-grow bg-primary-800 ring-2 ring-primary-900 rounded focus:outline-none"
    id="name-0"
    name="name-0"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="flex items-center mb-3.5"
  >
    <label
      className="font-semibold mr-auto"
      htmlFor="console-0"
    >
      Console*
    </label>
  </div>
  <select
    className="appearance-none w-full h-11 px-2 mb-5 flex-grow bg-primary-800 ring-2 ring-primary-900 rounded focus:outline-none"
    id="console-0"
    name="console-0"
    onChange={[Function]}
    value="NES"
  >
    <option
      value="NES"
    >
      NES
    </option>
    <option
      value="GB"
    >
      Game Boy
    </option>
    <option
      value="GBC"
    >
      Game Boy Color
    </option>
    <option
      value="CHIP 8"
    >
      CHIP 8
    </option>
  </select>
  <div
    className="flex mx-auto"
  >
    <label
      className="btn-primary h-7 w-24 mr-10"
      htmlFor="submit-0"
    >
      <input
        className="hidden"
        id="submit-0"
        name="submit-0"
        type="submit"
      />
      <svg
        className="inline-block mr-1.5"
        fill="currentColor"
        height="0.625rem"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="0.625rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
      <span
        className="font-medium"
      >
        Add
      </span>
    </label>
    <button
      className="font-medium text-primary-100 md:hover:text-red-500 active:text-red-500"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</form>
`;
